rollout/controller.go

```diff
@@ -14,9 +14,11 @@ import (
 	"k8s.io/client-go/dynamic"
 	appsinformers "k8s.io/client-go/informers/apps/v1"
 	coreinformers "k8s.io/client-go/informers/core/v1"
+	extensionsinformers "k8s.io/client-go/informers/extensions/v1beta1"
 	"k8s.io/client-go/kubernetes"
 	appslisters "k8s.io/client-go/listers/apps/v1"
 	v1 "k8s.io/client-go/listers/core/v1"
+	extensionslisters "k8s.io/client-go/listers/extensions/v1beta1"
```

Comment:
Agree. extensions/v1beta1 will no longer be served in the next v1.18 Kubernetes release. 

See [v1.14 changelog](https://v1-14.docs.kubernetes.io/docs/setup/release/notes/#deprecations)

  - Ingress resources are now available via `networking.k8s.io/v1beta1`. Ingress resources in `extensions/v1beta1` are deprecated and will no longer be served in v1.18. Existing persisted data is available via the new API group/version  ([#74057](https://github.com/kubernetes/kubernetes/pull/74057), [@liggitt](https://github.com/liggitt))

I think we can simply use the `networking.k8s.io/v1beta1` informer. We probably don't need to go so far as we did with with Istio VirtualService with the CLI flag to specify apiVerision, since the go-client might drop that package eventually.

Reply:
Caveat: if we don't provide the CLI flag, it means we do not support K8s v1.13.

Reply:
Yeah, `networking.k8s.io/v1beta` should be the default

Reply:
Sounds like a plan. I hope that the k8s v1.18 go-client SDK will still have the type definitions/informers libraries for `extensions/v1beta` even if the API server no longer serves it.

Reply:
tl;dr should we hold this back for kube 1.18 support?

I’ll see how this can be achieved without too much code duplication. I am still heavily leaning towards just supporting `extensions/v1beta1`. I tried to find a decent break down of kube versions in the wild but the only thing I found is that EKS just started supporting 1.14 a short while ago. So at least for AWS, 1.18 seems quite far out.
I know Argo projects probably don’t yet advertise a support matrix but maybe we should start?

Reply:
Support for `extensions/v1beta1` would be much appreciated. I'm in the same boat as @moensch being stuck with a 1.13 k8s distribution for the foreseeable future and have a number of teams looking forward for traffic shaped canary rollouts.

Reply:
To close the loop here from the slack conversation: Maintainers agreed to add kube 1.18 support in a later release.